eth/v1alpha1/validator.proto

```diff
@@ -48,6 +48,78 @@ service BeaconNodeValidator {
         };
     }
 
+    // DomainData fetches the current BLS signature domain version information from the
+    // running beacon node's state. This information is used when validators sign
+    // blocks and attestations appropriately based on their duty.
+    rpc DomainData(DomainRequest) returns (DomainResponse) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/domain"
+        };
+    }
+
+    // WaitForActivation checks if a validator public key exists in the active validator 
+    // registry of the current beacon state. If the validator is NOT yet active, it starts a 
+    // server-side stream which updates the client whenever the validator becomes active in
+    // the beacon node's state.
+    //
+    // The input to this endpoint is a list of validator public keys, and the corresponding
+    // stream will respond until at least a single corresponding validator to those
+    // keys is activated.
+    rpc WaitForActivation(ValidatorActivationRequest) returns (stream ValidatorActivationResponse) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/activation/stream"
+        };
+    }
+
+    // WaitForChainStart queries the logs of the Validator Deposit Contract on the Ethereum
+    // proof-of-work chain to verify the beacon chain has started its runtime and 
+    // validators are ready to begin their responsibilities. 
+    //
+    // If the chain has not yet started, this endpoint starts a server-side stream which updates
+    // the client when the beacon chain is ready.
+    rpc WaitForChainStart(google.protobuf.Empty) returns (stream ChainStartResponse) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/chainstart/stream"
+        };
+    }
+
+    // ValidatorIndex retrieves a validator's index location in the beacon state's
+    // validator registry looking up whether the validator exists based on its
+    // public key. This method returns NOT_FOUND if no index is found for the public key
+    // specified in the request.
+    rpc ValidatorIndex(ValidatorIndexRequest) returns (ValidatorIndexResponse) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/index"
+        };
+    }
+
+    // ValidatorPerformance reports a validator's latest balance along with other important 
+    // metrics on rewards and penalties throughout its lifecycle in the beacon chain.
+    // The request takes in a list of validator public keys and returns a performance report
+    // for all of them respectively.
+    rpc ValidatorPerformance(ValidatorPerformanceRequest) returns (ValidatorPerformanceResponse) {
```

Comment:
This is more of an info method rather than a requirement to serve validators. Should this go into another service?

Reply:
Good point, maybe in the beacon chain service